system/jlib/jtrace.cpp

```diff
@@ -735,6 +735,38 @@ class CSpan : public CInterfaceOf<ISpan>
         return span ? span->IsRecording() : false;
     }
 
+    virtual void recordSuccess(const char * successMessage) override
+    {
+        if (span != nullptr)
+        {
+            span->SetStatus(opentelemetry::trace::StatusCode::kOk, successMessage);
+        }
+    }
+
+    virtual void recordError(const char * errorMessage, int errorCode) override
+    {
+        if (span != nullptr)
+        {
+            span->SetStatus(opentelemetry::trace::StatusCode::kError, errorMessage);
+            if (errorCode != UNKNOWN_ERROR_CODE)
+                span->SetAttribute("error.code", errorCode);
+        }
+    }
+
+    virtual void recordException(IException * e)
+    {
+        if (span != nullptr)
+        {
+            StringBuffer msg;
+            e->errorMessage(msg);
+            recordError(msg.str(), e->errorCode());
+
+            span->AddEvent("Exception", {{"message", msg.str()}, {"type", typeid(e).name()}});
```

Comment:
Agreed on first point.
I agree on the second point as well, but thought it would be pedantic to force a "record as failure" flag, maybe I'll add it and default it, and allow the developer to reportexcpetion but not affirm failure.